test/e2e/tests/prague/upgrade-account.spec.ts

```diff
@@ -0,0 +1,109 @@
+                      address: '0xCd8D6C5554e209Fbb0deC797C6293cf7eAE13454',
+                    },
+                  ],
+                  '0x539': [
+                    {
+                      address: '0x8438Ad1C834623CfF278AB6829a248E37C2D7E3f',
+                      signature:
+                        '0x4c15775d0c6d5bd37a7aa7aafc62e85597ea705024581b8b5cb0edccc4e6a69e26c495b3ae725815a377c9789bff43bf19e4dd1eaa679e65133e49ceee3ea87f1b',
+                    },
+                  ],
+                },
+                supportedChains: ['0xaa36a7', '0x539'],
+              },
+            },
+          ],
+        };
+      }),
+  ];
+}
+
+describe('Upgrade Account', function (this: Suite) {
+  it('an EOA account can be upgraded when triggering a batch tx from a dapp in an odd chain id', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        localNodeOptions: [
+          {
+            type: 'anvil',
+            options: {
+              hardfork: 'prague',
+              loadState:
+                './test/e2e/seeder/network-states/eip7702-state/withDelegatorContracts.json',
+            },
+          },
+        ],
+        testSpecificMock: mockEip7702FeatureFlag,
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver }: { driver: Driver }) => {
+        await loginWithBalanceValidation(driver);
+
+        const testDapp = new TestDapp(driver);
+        await testDapp.openTestDappPage();
+        await testDapp.clickSendCalls();
+
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+        const upgradeAndBatchTxConfirmation = new Eip7702AndSendCalls(driver);
+
+        await upgradeAndBatchTxConfirmation.check_expectedTxTypeIsDisplayed(
+          'Smart account',
+        );
+        await upgradeAndBatchTxConfirmation.check_expectedInteractingWithIsDisplayed(
+          'Account 1',
+        );
+
+        // Open Settings and very tx details
+        await upgradeAndBatchTxConfirmation.openSettings();
+        await upgradeAndBatchTxConfirmation.check_batchTxListIsPresent();
+
+        // Confirm upgrade and batch tx
+        await upgradeAndBatchTxConfirmation.tickUpgradeCheckbox();
+        await upgradeAndBatchTxConfirmation.confirmUpgradeAndBatchTx();
+
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+
+        // Check the batch transaction has been successfully performed
+        // which included x2 simple sends of 0.001ETH, so the ETH balance should also be updated
+        const homePage = new HomePage(driver);
+        await homePage.goToActivityList();
+
+        const activityList = new ActivityListPage(driver);
+        await activityList.check_confirmedTxNumberDisplayedInActivity(1);
+        await homePage.check_expectedBalanceIsDisplayed('24.9998', 'ETH');
```

Comment:
In addition to checking the transaction is confirmed, could we also verify the result of `eth_getCode` on the upgraded account?

Reply:
great point! done :+1: